### PR TITLE
Add invert option to dearrow plugin

### LINKED ISSUE #2783: [FEATURE REQUEST - Dearrow] Setting for the default state
### Content

A setting could be added to either enable dearrow on youtube embeds by default (current behavior) or disable it by default (and clicking on the button per embed)

### Request Agreement

- [X] I have read the requirements for opening an issue above

### PR DESCRIPTION
**Pull request by electricsteve**
Fix #2783 
Just need feedback if the option should be marked restart needed. (line 174)

### CODE REVIEW

**Comment by Vendicated** [comment]
why do you need feedback for if a restart is needed? it simply depends on if the option can apply immediately or needs a restart to work

**Comment by EDM115** [comment]
@Vendicated from what I understand the option will have an immediate effect but will not work retroactively on already dearrowed embeds, unless the user reload their client

**Comment by electricsteve** [comment]
Yes, its just idk if it is worth requiring a restart.

**Comment by Vendicated** [comment]
ah i see. that's fine

**Comment by electricsteve** [comment]
Yea ok but my question is still not answered, @EDM115 what do you think?

**Comment by EDM115** [comment]
I believe it's alright, if users want to see changes on their old embeds that are already dearrowed they just have to refresh discord or close & reopen, which isn't a big problem  
I don't think that this setting needs a restart to work out of the box

**Comment by electricsteve** [comment]
Ok, will remove comment in a few min, currently eating.

**Comment by Nuckyz** [comment]
Instead of being an option to invert we should do something like how ShowHiddenChannels does:
image

Every time you click the toggle button it also updates the setting, so it remembers what you last chose

**Comment by electricsteve** [comment]
Wtf is that option name

**Comment by electricsteve** [comment]
> Instead of being an option to invert we should do something like how ShowHiddenChannels does: image
> 
> Every time you click the toggle button it also updates the setting, so it remembers what you last chose

OK will do that

**Comment by electricsteve** [comment]
But wait the name of that setting contradicts the function, because it isn't the default, right?

**Comment by electricsteve** [comment]
(Spelling mistake)

**Comment by Nuckyz** [comment]
> But wait the name of that setting contradicts the function, because it isn't the default, right?

It's the default value which is applied the next time you view the button where it's used

File: src/plugins/dearrow/index.tsx [hunk 1]
```diff
@@ -95,7 +108,9 @@ function DearrowButton({ component }: { component: Component<Props>; }) {
                     onMouseLeave={onMouseLeave}
                     className={"vc-dearrow-toggle-" + (embed.dearrow.enabled ? "on" : "off")}
                     onClick={() => {
+                        const { invert } = settings.store;
```
Comment (Nuckyz) [inline src/plugins/dearrow/index.tsx:111]:
```suggestion
```

File: src/plugins/dearrow/index.tsx [hunk 2]
```diff
@@ -95,7 +108,9 @@ function DearrowButton({ component }: { component: Component<Props>; }) {
...
                         const { enabled, oldThumb, oldTitle } = embed.dearrow;
+                        settings.store.invert = !invert;
```
Comment (Nuckyz) [inline src/plugins/dearrow/index.tsx:113]:
```suggestion
                        settings.store.invert = !enabled;
```

**Review by Nuckyz — CHANGES_REQUESTED** [review]

File: src/plugins/dearrow/index.tsx [hunk 3]
```diff
@@ -77,6 +77,19 @@ async function embedDidMount(this: Component<Props>) {
...
+            embed.dearrow.enabled = !enabled;
+            if (oldTitle) {
+                embed.dearrow.oldTitle = embed.rawTitle;
+                embed.rawTitle = oldTitle;
+            }
+            if (oldThumb) {
+                embed.dearrow.oldThumb = embed.thumbnail.proxyURL;
+                embed.thumbnail.proxyURL = oldThumb;
+            }
+        }
```
Comment (Vendicated) [inline src/plugins/dearrow/index.tsx:91]: this is just undoing the previous work. why not just use a condition above?

File: src/plugins/dearrow/index.tsx (hunk 2 above)
Comment (electricsteve) [reply src/plugins/dearrow/index.tsx:113]: Ah ok

File: src/plugins/dearrow/index.tsx (hunk 3 above)
Comment (electricsteve) [reply src/plugins/dearrow/index.tsx:91]: Will do!

**Comment by Nuckyz** [comment]
Thank you for the contribution!

**Comment by Nuckyz** [comment]
> NP! also I didn't notice enabled didn't need to be inverted. LOL

no worries, my suggestion was wrong too, I only noticed after testing haha

**Pull request merged**